src/dos/dos.cpp

```diff
@@ -1204,6 +1204,25 @@ static Bitu DOS_26Handler(void) {
     return CBRET_NONE;
 }
 
+void set_ver(char *args, bool start)
+{
+	char *word = StripWord(args);
+	if (!*args && !*word) { // Reset
+		dos.version.major = 5;
+		dos.version.minor = 0;
+	} else if (*args == 0 && *word && (strchr(word, '.') != 0)) { // Allow:
+		                                                      // ver set 5.1
+		const char *p = strchr(word, '.');
+		dos.version.major = (Bit8u)(atoi(word));
+		dos.version.minor = (Bit8u)(strlen(p + 1) == 1 && *(p + 1) > '0' &&
+		                                            *(p + 1) <= '9'
+		                                    ? atoi(p + 1) * 10
+		                                    : atoi(p + 1));
```

Comment:
The right side is very hard to look at and understand, as someone who isn't familiar with what's going on.
Can it be split up into separate lines of logic, storing intermediate values in nicely described variables?

Reply:
@kcgen I will rework on this part of code tomorrow to make it easier to understand.

Reply:
Please check out the new code - there are now multiple lines for setting dos.version.minor with comments added.